File: 125-valid-palindrome/valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    s = s.toLowerCase()
    s = s.replace(/[^0-9a-zA-Z]/g, "")
    if (s === "") {
        return true
    }
    let l = 0
    let r = s.length - 1
    while (l < r) {
        if (s[l] === s[r]) {
            l++
            r--
        } else {
            return false
        }
    }
    return true

    
};

File: 125-valid-palindrome/valid-palindrome.rb
# @param {String} s
# @return {Boolean}
def is_palindrome(s)
    # Inputs: given a string s 
    # Output: true or false whether s is a palindrome or not
    normalized_string = s.downcase.gsub(/[^a-z0-9]/, "")
    normalized_string == normalized_string.reverse
end